scripts/idl/zapxml/handlers/handlers.py

```diff
@@ -0,0 +1,617 @@
+        self._command = None
+        self._struct = Struct(name=attrs['name'], fields=[])
+        self._field_index = 0  # commands DO NOT support field index it seems
+
+        if attrs['source'].lower() == 'client':
+            self._struct.tag = StructTag.REQUEST
+
+            name = attrs['name']
+
+            if name.endswith('Request'):
+                request_name = name
+                command_name = name[:-7]
+            else:
+                request_name = name+'Request'
+                command_name = name
+
+            self._struct.name = request_name
+
+            if 'response' in attrs:
+                response_name = attrs['response']
+            else:
+                response_name = 'DefaultResponse'
+
+            # TODO: what if no response? DefaultResponse?
+            self._command = Command(
+                name=name,
+                code=ParseInt(attrs['code']),
+                input_param=request_name,
+                output_param=response_name,
+            )
+
+            if attrs.get('isFabricScoped', 'false') == 'true':
+                self._command.attributes.add(CommandAttribute.FABRIC_SCOPED)
+
+            if attrs.get('mustUseTimedInvoke', 'false') == 'true':
+                self._command.attributes.add(CommandAttribute.TIMED_INVOKE)
+
+        else:
+            self._struct.tag = StructTag.RESPONSE
+            self._struct.code = ParseInt(attrs['code'])
+
+    def GetArgumentField(self, attrs):
+        data_type = DataType(name=attrs['type'])
+
+        if 'length' in attrs:
+            data_type.max_length = ParseInt(attrs['length'])
+
+        self._field_index += 1
+
+        field = Field(
+            data_type=data_type,
+            code=self._field_index,
+            name=attrs['name'],
+            is_list=(attrs.get('array', 'false') == 'true')
+        )
+
+        if attrs.get('optional', "false").lower() == 'true':
+            field.attributes.add(FieldAttribute.OPTIONAL)
+
+        if attrs.get('isNullable', "false").lower() == 'true':
+            field.attributes.add(FieldAttribute.NULLABLE)
+
+        return field
+
+    def GetNextProcessor(self, name: str, attrs):
+        if name.lower() == 'access':
+            if attrs['op'] != 'invoke':
+                raise Exception('Unknown access for %r' % self._struct)
+
+            if self._command:
+                self._command.invokeacl = AttrsToAccessPrivilege(attrs)
+            else:
+                logging.warning("Ignored access role for reply %r" % self._struct)
+            return BaseHandler(self.context, handled=HandledDepth.SINGLE_TAG)
+        elif name.lower() == 'arg':
+            self._struct.fields.append(self.GetArgumentField(attrs))
+            return BaseHandler(self.context, handled=HandledDepth.SINGLE_TAG)
+        elif name.lower() == 'description':
+            return BaseHandler(self.context, handled=HandledDepth.ENTIRE_TREE)
+        else:
+            # TODO: implement
```

Comment:
Please file an issue and reference here with `TODO(#issue):`

Reply:
Removed todo - this was implemented